Mathlib/NumberTheory/SumPrimeReciprocals.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2023 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.NumberTheory.SmoothNumbers
+import Mathlib.Analysis.PSeries
+
+/-!
+# The sum of the reciprocals of the primes diverges
+
+We show that the sum of `1/p`, where `p` runs through the prime numbers, diverges.
+We follow the elementary proof by Erdős that is reproduced in "Proofs from THE BOOK".
+There are two versions of the main result: `not_summable_one_div_on_primes`, which
+expresses the sum as a sub-sum of the harmonic series, and `Nat.Primes.not_summable_one_div`,
+which writes it as a sum over `Nat.Primes`. We also show that the sum of `p^r` for `r : ℝ`
+converges if and only if `r < -1`; see `Nat.Primes.summable_rpow`.
+
+## References
+
+See the sixth proof for the infinity of of primes in Chapter 1 of [aigner1999proofs].
+The proof is due to Erdős.
+-/
+
+/-- The cardinality of the set of `k`-rough numbers `≤ N` is bounded by `N` times the sum
+of `1/p` over the primes `k ≤ p ≤ N`. -/
+-- This needs `Mathlib.Data.IsROrC.Basic`, so we put it here
+-- instead of in `Mathlib.NumberTheroy.SmoothNumbers`.
+lemma Nat.roughNumbersUpTo_card_le' (N k : ℕ) :
+    (roughNumbersUpTo N k).card ≤
+      N * (N.succ.primesBelow \ k.primesBelow).sum (fun p ↦ (1 : ℝ) / p) := by
+  simp_rw [Finset.mul_sum, mul_one_div]
+  exact (Nat.cast_le.mpr <| roughNumbersUpTo_card_le N k).trans <|
+    (cast_sum (β := ℝ) ..) ▸ Finset.sum_le_sum fun n _ ↦ cast_div_le
+
+open Set Nat BigOperators
+
+/-- The sum over primes `k ≤ p ≤ 4^(π(k-1)+1)` over `1/p` (as a real number) is at least `1/2`. -/
+lemma sum_primes_ge_one_div_ge_half (k : ℕ) :
+    ∑ p in (4 ^ (k.primesBelow.card + 1)).succ.primesBelow \ k.primesBelow,
+      (1 / p : ℝ) ≥ 1 / 2 := by
+  set m : ℕ := 2 ^ k.primesBelow.card
+  set N₀ : ℕ := 2 * m ^ 2 with hN₀
+  let S : ℝ := ((2 * N₀).succ.primesBelow \ k.primesBelow).sum (fun p ↦ (1 / p : ℝ))
+  suffices : 2 * N₀ ≤ m * (2 * N₀).sqrt + 2 * N₀ * S
+  · rw [ge_iff_le]
+    rw [hN₀, ← mul_assoc, ← pow_two 2, ← mul_pow, sqrt_eq', ← sub_le_iff_le_add'] at this
+    push_cast at this
+    ring_nf at this
```

Comment:
This seems quite annoying to maintain as the ring_nf output could randomly change. Can you spell out the equality you want
like
```
rw [show a * b = b * a by ring] at this
```

Reply:
OK; changed the proof.